### PR TITLE
New Paddle-CINN Compile PR

### PR DESCRIPTION
**Pull request by zhhsplendid**
### PR types

Bug fixes
### PR changes

Others
### Describe

This PR added some changes to match the CINN change for compilation. It also tried to fix JiangCheng's Problem in PR: https://github.com/PaddlePaddle/Paddle/pull/36100

These changes include:
1. Set `CINN_GIT_TAG` to a newer tag
2. CINN now just `make cinnapi -j`
3. We have to add `-DPY_VERSION=${PY_VERSION} -DWITH_TESTING=ON` to CINN cmake args
4. For CINN's third party dependencies, we could just include headers without target_link_libraries
5. Moved `cinn.cmake` from `paddle/cmake` to `paddle/cmake/external` to match old style. External folder contains `lite`, which is the same level of `cinn`
6. CINN added `-DNAMESPACE=cinn_gflags` in `gflags.cmake` to have different gflag namespaces between CINN and Paddle. It solved re-define problem.
7. Change namespace of `::google::` in gflags to `::GFLAGS_NAMESPACE`

### CODE REVIEW

File: paddle/fluid/inference/tests/infer_ut/test_LeViT.cc [hunk 1]
```diff
@@ -174,6 +174,10 @@ TEST(tensorrt_tester_LeViT, multi_thread4_trt_fp32_bz2) {
 
 int main(int argc, char** argv) {
   ::testing::InitGoogleTest(&argc, argv);
+#ifdef GFLAGS_NAMESPACE
+  GFLAGS_NAMESPACE::ParseCommandLineFlags(&argc, &argv, true);
+#else
```
Comment (wzzju) [inline paddle/fluid/inference/tests/infer_ut/test_LeViT.cc]:
```suggestion
::GFLAGS_NAMESPACE::ParseCommandLineFlags(&argc, &argv, true);
```
直接写成这样就可以了，因为GFLAGS_NAMESPACE默认定义就是`google`。其他的地方也是类似。

Comment (zhhsplendid) [reply paddle/fluid/inference/tests/infer_ut/test_LeViT.cc]: Done

**Review by wzzju — DISMISSED** [review]
LGTM.

**Review by wanghuancoder — APPROVED** [review]
LGTM

**Review by wzzju — APPROVED** [review]
LGTM.

**Pull request merged**